src/main/java/org/isf/medicalinventory/manager/MedicalInventoryManager.java

```diff
@@ -372,4 +386,55 @@ public void deleteInventory(MedicalInventory medicalInventory) throws OHServiceE
 		}
 		ioOperations.deleteInventory(medicalInventory);
 	}
+
+	/**
+	 * Confirm the Inventory rows of inventory.
+	 *
+	 * @param inventory - The {@link MedicalInventory}
+	 * @param inventoryRowSearchList- The list of {@link MedicalInventory}
+	 * @throws OHDataValidationException
+	 */
+	@Transactional(rollbackFor = OHServiceException.class)
+	public void confirmMedicalInventoryRow(MedicalInventory inventory, List<MedicalInventoryRow> inventoryRowSearchList) throws OHServiceException {
+		// validate the inventory
+		this.validateMedicalInventoryRow(inventory, inventoryRowSearchList);
+		// create Movements
+		String dischargeCode = inventory.getDischargeType();
+		String chargeCode = inventory.getChargeType();
+		Integer supplierId = inventory.getSupplier();
+		String wardCode = inventory.getDestination();
+		MovementType chargeType = medicalDsrStockMovementTypeBrowserManager.getMovementType(chargeCode);
+		MovementType dischargeType = medicalDsrStockMovementTypeBrowserManager.getMovementType(dischargeCode);
+		Supplier supplier = supplierManager.getByID(supplierId);
+		Ward ward = wardManager.findWard(wardCode);
+		LocalDateTime now = TimeTools.getNow();
+		for (Iterator<MedicalInventoryRow> iterator = inventoryRowSearchList.iterator(); iterator.hasNext();) {
+			MedicalInventoryRow medicalInventoryRow = (MedicalInventoryRow) iterator.next();
+			StringBuilder referenceBuilder = new StringBuilder();
+			referenceBuilder.append(inventory.getInventoryReference())
+			                .append("-")
+			                .append(now);
+			double theoQty = medicalInventoryRow.getTheoreticQty();
+			double realQty = medicalInventoryRow.getRealQty();
+			Double ajustQty = theoQty - realQty;
+			Medical medical = medicalInventoryRow.getMedical();
+			String lotCode = medicalInventoryRow.getLot().getCode();
+			Lot currentLot = movStockInsertingManager.getLot(lotCode);
+			if (realQty > theoQty) { // charge movement when realQty > theoQty
+				referenceBuilder.append("-charge");
+				String reference = referenceBuilder.toString();
+				Movement movement = new Movement(medical, chargeType, null, currentLot, now, -(ajustQty.intValue()), supplier, reference);
+				List<Movement> chargeMovement = new ArrayList<>();
+				chargeMovement.add(movement);
+				chargeMovement = movStockInsertingManager.newMultipleChargingMovements(chargeMovement, reference);
+			} else if (realQty < theoQty) { // discharge movement when realQty < theoQty
+				referenceBuilder.append("-discharge");
+				String reference = referenceBuilder.toString();
+				Movement movement = new Movement(medical, dischargeType, ward, currentLot, now, ajustQty.intValue(), null, reference);
+				List<Movement> dischargeMovement = new ArrayList<>();
+				dischargeMovement.add(movement);
+				dischargeMovement = movStockInsertingManager.newMultipleDischargingMovements(dischargeMovement, reference);
```

Comment:
you are inserting only one movement... this should be likely called after the iteration cycle, and only if chargeMovement is not empty

```suggestion
				dischargeMovement = movStockInsertingManager.newMultipleDischargingMovements(dischargeMovement, reference);
```